File: Selection sort/SelectionSort.php
<?php

class SelectionSort
{
    public function selectionSorter($array)
    {
        $unsortedSize = count($array);
        while ($unsortedSize > 1) {
            $maxPos = 0;
            for ($i = 0; $i < $unsortedSize; $i++) {
                if ($array[$i] >= $array[$maxPos])
                    $maxPos = $i;
            }
            $unsortedMax = $array[$maxPos];
            $array[$maxPos] = $array[--$unsortedSize];
            $array[$unsortedSize] = $unsortedMax;
        }
        return $array;
    }
}

File: Selection sort/selection_sort.py
def selection_sort(array: list) -> list:
    """Sorts a list of numbers in ascending order using selection sort algorithm"""
    unsorted_size = len(array)
    while unsorted_size > 0:
        max_pos = 0
        for j in range(unsorted_size):
            if array[j] >= array[max_pos]:
                max_pos = j
        unsorted_size -= 1
        temp_max = array[max_pos]
        array[max_pos] = array[unsorted_size]
        array[unsorted_size] = temp_max
    return array


pi = [3, 1, 4, 1, 5, 9, 2, 6, 5, 3, 5, 8, 9, 7, 9, 3, 2, 3, 8, 4, 6]
print(selection_sort(pi))
